flags.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+)
+
+type kwargStringFlag struct {
+	value   *map[string][]byte
+	changed bool
+}
+
+func newKwargStringFlag() *kwargStringFlag {
+	return &kwargStringFlag{
+		value: new(map[string][]byte),
+	}
+}
+
+func (f *kwargStringFlag) AsMap() map[string][]byte {
+	var m map[string][]byte
+	if f.value != nil {
+		return *f.value
+	}
+	return m
+}
+
+func (f *kwargStringFlag) String() string {
+	m := f.AsMap()
+	return fmt.Sprintf("%v", m)
+}
+
+func (f *kwargStringFlag) Set(input string) error {
+	p, err := parseKeyValuePair(input)
+	if err != nil {
+		return err
+	}
+	if !f.changed {
+		// first time set is called so assign to *f.value
+		*f.value = map[string][]byte{p.Key: p.Value}
+	} else {
+		// after first time just update existing *f.value
+		(*f.value)[p.Key] = p.Value
+	}
+	f.changed = true
+	return nil
+}
+
+func (f *kwargStringFlag) Type() string {
```

Comment:
are Set() and Type() implementing an interface? If so force the compiler to enforce it `var _ pflag.WhateverInterface = kwargStringFlag{}`

Reply:
Yep. Can do that.